### PR TITLE
Claire.js: refactored again + adding/removing dependencies.

### PR DESCRIPTION
**Pull request by mzhurovich**
CC @sompylasar

### CODE REVIEW

File: Karl/integrations/nodejs/test.js [hunk 1]
```diff
@@ -83,8 +83,44 @@ describe('Claire JS client test.', function () {
...
+          assert.equal(claire.dependencies.length, 1);
+          assert.equal(claire.dependencies[0].ip, '127.0.0.1');
+          assert.equal(claire.dependencies[0].port, 8283);
+          assert.equal(claire.dependencies[0].prefix, '/');
+          completed = true;
+        });
+      });
+    });
+
+    var wait = function () {
```
Comment (dkorolev) [inline Karl/integrations/nodejs/test.js]: Just call `done();` from where you set `completed = true` above. Also, IIFE ftw:

``
(function(i) {
  console.log(i);
})(42);

File: Karl/integrations/nodejs/claire.js [hunk 2]
```diff
@@ -80,128 +79,186 @@ function Claire (config) {
...
       if (typeof callback === 'function') {
         if (error) {
-          callback({ success: false, error: 'HTTP connection attempt failed' });
+          callback({ message: 'HTTP connection attempt failed' });
         } else if (response.statusCode < 200 || response.statusCode > 299) {
-          callback({ success: false, error: 'HTTP request error', response: response });
+          callback({ message: 'HTTP request error', http_response: response });
         } else {
-          callback({ success: true });
+          callback(null);
```
Comment (sompylasar) [inline Karl/integrations/nodejs/claire.js]: :+1:

File: Karl/integrations/nodejs/claire.js [hunk 3]
```diff
@@ -80,128 +79,186 @@ function Claire (config) {
...
 
+Claire.prototype.addDependency = function (dependencyUrl, callback) {
+  var that = this;
+  this.makeDependencyKey_(dependencyUrl, function (error, key) {
+    if (error) {
+      callback(error);
+    } else {
+      if (that.findDependency_(key) === undefined) {
+        that.dependencies_.push(key);
+        callback(null, key);
```
Comment (sompylasar) [inline Karl/integrations/nodejs/claire.js]: `callback(null, { key: key })`

Comment (sompylasar) [inline Karl/integrations/nodejs/claire.js]: Okay. Forget this.

File: Karl/integrations/nodejs/claire.js [hunk 4]
```diff
@@ -80,128 +79,186 @@ function Claire (config) {
...
     // Clear existing keepalive timeout and restart the loop.
-    this.clearKeepaliveTimeout();
-    this.keepaliveLoop();
+    this.clearKeepaliveTimeout_();
+    this.keepaliveLoop_();
+  }
+};
+
+Claire.prototype.makeDependencyKey_ = function (dependencyUrl, callback) {
+  var parsedUrl = url.parse(dependencyUrl);
```
Comment (sompylasar) [inline Karl/integrations/nodejs/claire.js]: `parsedUrl` -> `dependencyUrlParts` or `dependencyUrlParsed`

File: Karl/integrations/nodejs/test.js (hunk 1 above)
Comment (sompylasar) [inline Karl/integrations/nodejs/test.js]: :+1: re: done()

Confused re: IIFE here.

Comment (dkorolev) [inline Karl/integrations/nodejs/test.js]: Unnecessary here, but helps saving variable name when it's unnecessary,

Comment (sompylasar) [inline Karl/integrations/nodejs/test.js]: But the variable name is used here to reference this function in the setTimeout call.

**Comment by dkorolev** [comment]
@sompylasar Thanks for the review!
@mzhurovich I'll merge it to not have the PR hanging. Bits pointed out by Ivan are worth addressing. :-)

Thanks!
Dima

**Pull request merged**